esphome: 2022.8.3 -> 2022.9.1

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for esphome is: "Make creating custom firmwares for ESP32/ESP8266 super easy"

meta.homepage for esphome is: "https://esphome.io/"

meta.changelog for esphome is: ""


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 2022.9.0 with grep in /nix/store/h9jn4apmjr216asz3xa8wp9fys7i4i2k-esphome-2022.9.0
- found 2022.9.0 in filename of file in /nix/store/h9jn4apmjr216asz3xa8wp9fys7i4i2k-esphome-2022.9.0

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
1 total rebuild path(s)

1 package rebuild(s)

First fifty rebuilds by attrpath
esphome
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/h9jn4apmjr216asz3xa8wp9fys7i4i2k-esphome-2022.9.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A esphome https://github.com/r-ryantm/nixpkgs/archive/5b57279aceab3c86978b6f02693c5a4e6ed01da2.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/h9jn4apmjr216asz3xa8wp9fys7i4i2k-esphome-2022.9.0
ls -la /nix/store/h9jn4apmjr216asz3xa8wp9fys7i4i2k-esphome-2022.9.0/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>esphome</li>
  </ul>
</details>

---

###### Maintainer pings

cc @globin @mweinelt for [testing](https://github.com/ryantm/nixpkgs-update/blob/master/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

mweinelt commented:
Please test esp8266 and esp32 builds with esphome before merging major esphome upgrades.

mweinelt reviewed (changes requested):
Missing dashboard updates.

https://github.com/esphome/esphome/releases/tag/2022.9.0

mweinelt committed:
esphome: 2022.8.3 -> 2022.9.1

https://github.com/esphome/esphome/releases/tag/2022.9.0
https://github.com/esphome/esphome/releases/tag/2022.9.1

mweinelt commented:
Tested on ESP8266.
